### ISSUE TITLE
onRowsDelete not functioning properly as an async function

### ISSUE THREAD

**Issue by T-kON99** [gregnb/mui-datatables#1277]
When handling onRowsDelete and returning false in an async function, the selected row still gets deleted even though it shouldn't

## Expected Behavior

Returning false in an async function should prevent deletion of the selected row as the documentation stated

## Current Behavior

Returning false in an async function still deletes the selected row anyway triggering the re-render state of the table.

## Steps to Reproduce (for bugs)

    handleDelete = async (rowsDeleted) => {
        return false;
    };
    render() {
        return(
            <MUIDataTable 
                data={this.state.eventDetails.attendants}
                columns={columns}
                options={{
                    onRowsDelete: this.handleDelete,
                    responsive: 'scrollMaxHeight',
                    searchPlaceholder: 'Search...'
                }}
            />
        );
    };

Will result in MUIDataTable deleting the selected row even though the return value is false.

When the handleDelete is not an async function, it works as expected.

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI-Core  |    ^4.9.9     |
| Material-UI-Icon |    ^4.9.1    |
| MUI-datatables  |    ^2.14.0     |
| React        |     ^16.12.0    |
| browser      |     Version 81.0.4044.129 (Official Build) (64-bit)    |
| etc          |     -    |

**Comment by patorjk** [comment]
There's an open PR for this feature to be async: https://github.com/gregnb/mui-datatables/pull/835

The main thing holding it up is getting async/await to transpile correctly for the library. 

As a workaround, a custom delete operation can be written using the customToolbarSelect option.

**Comment by T-kON99** [comment]
For now I switched from using async onRowsDelete to the sync version of it. I think #835 main issue is using MaterialUI for the design which is not transpiling correctly for the webpack is that right?

**Comment by patorjk** [comment]
Correct, the transpiling has issues with async/await. If that were resolved, onRowsDelete could be made async.
